src/main/kotlin/no/nav/syfo/behandler/BehandlerService.kt

```diff
@@ -33,4 +61,67 @@ class BehandlerService(
 
         return null
     }
+
+    private fun createOrGetBehandler(fastlege: Fastlege, personIdentNumber: PersonIdentNumber): Behandler {
+        val behandlerDialogmeldingForPersonList = database.getBehandlerDialogmeldingForPerson(
+            personIdentNumber = personIdentNumber,
+        )
+
+        val behandlerDialogmeldingForPersonsFastlege =
+            behandlerDialogmeldingForPersonList.find { (_, pBehandlerDialogmelding) -> pBehandlerDialogmelding.partnerId == fastlege.partnerId.toString() }
+
+        if (behandlerDialogmeldingForPersonsFastlege == null) {
+            return createBehandlerDialogmelding(
```

Comment:
Må du ikke først sjekke om behandleren finnes (dvs finnes, men ikke blant dem som er knyttet til personIdentNumber)?

Reply:
Det gjøres vel nå siden spørringen gjør left join og tar med seg også behandlere som ikke er knyttet til personident. Men som du skriver under er det nok bedre å ta partnerId inn i spørringen - skal se på det.

Reply:
Refaktorert til to spørringer og endret service tilsvarende nå.